updater/storage.go

```diff
@@ -71,29 +75,40 @@ func (reg *ResourceRegistry) ScanStorage(root string) error {
 	return lastError
 }
 
-// LoadIndexes loads the current release indexes from disk and will fetch a new version if not available and online.
+// LoadIndexes loads the current release indexes from disk
+// or will fetch a new version if not available and the
+// registry is marked as online.
 func (reg *ResourceRegistry) LoadIndexes() error {
-	err := reg.loadIndexFile("stable.json", true, false)
-	if err != nil {
-		err = reg.downloadIndex("stable.json", true, false)
-		if err != nil {
-			return err
+	var firstErr error
+	for _, idx := range reg.indexes {
```

Comment:
Shouldn't this also be using `reg.getIndexes()`?

Reply:
Yep, it should :)